### PR TITLE
[CAM-8439] - Document CAM CostCenters API endpoints

### PR DESCRIPTION
**Pull request by julio-castanar**
@gsanchezu 
Please, review this pull rquest

### CODE REVIEW

**Comment by julio-castanar** [comment]
Please, consider this is a new document with cost centers, there was no previous one as a reference. 
Several parameters in PUT command could be included or not but some of them could compromise security policy: clc_alias, account_installed_produc, billing_account_number.
Other parameters like icon_metadata are obtained from code. I don´t know if they must be shown in the API documentation.

**Comment by julio-castanar** [comment]
@robersantamaria
Please, could you check this PR especially the correct working of the scripts that appear with calls to the API ?

File: CAM Platform/api-cost-centers.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+| organization | string | Name of the organization with access to the Cost Center. | Yes |
+| schema | string | Cost Center schema. Always “http://elasticbox.net/schemas/costcenter” | Yes  |
+| members | array | Members' id list of the Cost Center as Administrators. | No |
+
+```
+{
+    "name": "Cost Center example",
+    "organization": "elasticboxdemo",
+    "schema": "http://elasticbox.net/schemas/costcenter",
+    "members":[{"id":"elasticboxdemo"}]
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: The documents in this array require 3 parameters: `id`, `role` and `type`. `role` will always be `administrator` and `type` can be `workpace` or `ldap`. `id` can then be a workspace ID or a ldap group DN.

File: CAM Platform/api-cost-centers.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+|------|------|------------|------|
+| name | string | Cost Center name. | Yes  |
+| organization | string | Name of the organization with access to the Cost Center. | Yes |
+| schema | string | Cost Center schema. Always “http://elasticbox.net/schemas/costcenter” | Yes  |
+| members | array | Members' id list of the Cost Center as Administrators. | No |
+
+```
+{
+    "name": "Cost Center example",
+    "organization": "elasticboxdemo",
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Better drop the _elastibox_ references where possible, switch to `camdemo` or just `demo_account`.

File: CAM Platform/api-cost-centers.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+Date: Thu, 27 Dec 2018 09:57:34 GMT
+Elasticbox-Release: 4.0.0
+```
+#### Response Parameters
+
+| Parameter | Type |Description |
+|-----------|------|------------|
+| clc_alias | string | Costcenter account alias in the organization. |
+| created | string | Creation date. Example “2015-07-02 10:23:47.748740” |
+| deleted | object | Identifies whether the Cost Center is deleted. |
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: _Identifies whether the Cost Center is deleted; `null` for existing objects_

File: CAM Platform/api-cost-centers.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+    "schema": "http://elasticbox.net/schemas/costcenter",
+    "updated": "2018-12-27 10:53:26.389183",
+    "uri": "/services/costcenters/5be2de28-1862a-3238-9a50d81"
+}
+```
+
+
+## GET /services/costcenters
+
+Fetches an existing Cost Center of the authenticated user.
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Drop the _ of the authenticated user_.

File: CAM Platform/api-cost-centers.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+
+### URL
+
+#### Structure
+```
+[GET] /services/costcenters/{costcenter_id}
+```
+#### Example
+```
+[GET] https://cam.ctl.io/services/costcenters/5be2de281862a32389a50d81
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Cost center IDs are UUID, use a fake but correctly formatter one.

File: CAM Platform/api-cost-centers.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    "deleted": null,
+    "uri": "/services/costcenters/5be2de28-1862a-3238-9a50d81",
+    "members": [
+        {
+            "type": "workspace",
+            "role": "administrator",
+            "id": "elasticboxdemo"
+        },
+    ],
+    "clc_alias": "A79A",
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Make sure this is a fake one.

File: CAM Platform/api-cost-centers.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+    "organization": "elasticboxdemo",
+    "id": "5be2de28-1862a-3238-9a50d81",
+    "schema": "http://elasticbox.net/schemas/costcenter"
+}
+```
+
+## PUT /services/costcenters
+
+Updates one or several values on an existing Cost Center of the authenticated user.
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Drop the _of the authenticated user_.

File: CAM Platform/api-cost-centers.md [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+
+### Response
+
+#### Normal Response Codes
+
+- **200** OK
+
+#### Common Error Response Codes
+- **400** Bad Request
+- **401** Unauthorized
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Add 404 Not Found

File: CAM Platform/api-cost-centers.md [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+
+| Name | Type | Description | Req. |
+|------| ---- | ----------- | ---- |
+| id | string | Cost Center unique identifier. | Yes |
+| members | array | Members' list of the Cost Center. | Yes |
+| name | string | Cost Center name. | Yes |
+| organization | string | Organization which the Cost Center belongs to. | Yes |
+| schema | string | Schema URI. http://elasticbox.net/schemas/costcenter. | Yes |
+| account_installed_product | string | Account Installed Product. | No |
+| billing_account_number | string | Billing account number. | No |
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Attributes like BAN and AIP cannot be edited by end users, even if they are admins.

File: CAM Platform/api-cost-centers.md [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+**Note:**
+Be careful using this command because some values like *members* will be added or deleted if they exists or not in the request parameter 
+
+```
+{
+    "id":"dbfdd7a7-c348-4d55-817a-5890699bbf86",
+    "name":"Cost Center example",
+    "organization":"elasticboxdemo",
+    "schema":"http://elasticbox.net/schemas/costcenter",
+    "members":[{"id":"elasticboxdemo"}]
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Use a full cost center object, like the one in the GET example, to show that this should include all parameters returned by a GET.

File: CAM Platform/api-cost-centers.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+```
+
+### Response
+
+#### Normal Response Codes
+
+- **200** OK
+
+#### Common Error Response Codes
+- **400** Bad Request
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Add 404 Not Found

File: CAM Platform/api-cost-centers.md [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+
+#### Common Error Response Codes
+- **401** Unauthorized
+- **404** Not Found
+
+#### Headers
+
+```
+Content-Type: application/json
+Content-Length: 489
```
Comment (robersantamaria) [inline CAM Platform/api-cost-centers.md]: Remove these headers and `Authorization`.

File: CAM Platform/api-cost-centers.md [hunk 13]
```diff
@@ -0,0 +1,337 @@
+{{{
+"title": "Cost Centers API",
+"date": "12-19-2018",
+"author": "Julio Castanar",
+"attachments": [],
+"contentIsHTML": false,
+"keywords": ["api", "Cost center"]
```
Comment (gsanchezu) [inline CAM Platform/api-cost-centers.md]:
```suggestion
"keywords": ["cam', "cloud application manager", "api", "Cost center"]
```

File: CAM Platform/api-cost-centers.md (hunk 10 above)
Comment (julio-castanar) [reply CAM Platform/api-cost-centers.md]: Parameters not shown must not be provided. They are not updated when requesting this command

File: CAM Platform/api-cost-centers.md [hunk 14]
```diff
@@ -0,0 +1,341 @@
...
+| role | string | always is *administrator*. | No |
+| type | string | can be *workspace* (default) or *ldap*. | No |
+
+
+```
+{
+    "name": "Cost Center example",
+    "organization": "camdemo",
+    "schema": "http://elasticbox.net/schemas/costcenter",
+    "members":[{"id":"camdemo"}]
```
[the remark this answers is not in the document]
Comment (julio-castanar) [reply CAM Platform/api-cost-centers.md]: API behavior is programmed to ignore those parameters, that's why I don't include them

File: CAM Platform/api-cost-centers.md [hunk 15]
```diff
@@ -0,0 +1,341 @@
...
+| uri | string | Cost Center URI. |
+
+#### Response Body
+
+```
+{
+    "created": "2018-12-27 10:53:26.389183",
+    "deleted": null,
+    "id": "1072ca63-d84d-42f2-87a5-f09c72338eb1",
+    "members": [{"role": "administrator","id": "camdemo"}],
```
[the remark this answers is not in the document]
Comment (julio-castanar) [reply CAM Platform/api-cost-centers.md]: Same as what we have commented

[the remark this answers is not in the document]
Comment (robersantamaria) [reply CAM Platform/api-cost-centers.md]: This is the response body, so this payload does include 3 parameters.

[the remark this answers is not in the document]
Comment (julio-castanar) [reply CAM Platform/api-cost-centers.md]: OK, Roberto. It has been updated in all response code

File: CAM Platform/api-cost-centers.md [hunk 16]
```diff
@@ -339,3 +345,13 @@ ElasticBox-Release: 4.0
 #### Response Body
 
 * None
+
+## Contacting Cloud Application Manager Support
```
Comment (gsanchezu) [inline CAM Platform/api-cost-centers.md]: Please, remove this section, it is not required on CAM API pages

**Pull request merged**